### PR TITLE
(enh) Dockerfile.j2: improve env vars for bash and activate in .bashrc

### PR DESCRIPTION
**Pull request by tobitege**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Enhance the Dockerfile.j2 template to set path correctly and activate virtual env in bash.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- added commands to bottom of template, only if "not skip_init", to set paths for virtual env in .bashrc
- replaced "od-" and "od_" with "oh"-pendants across the codebase, finally
- added some comments

---
**Link of any specific issues this addresses**

### CODE REVIEW

File: openhands/runtime/utils/runtime_templates/Dockerfile.j2 [hunk 1]
```diff
@@ -46,23 +47,44 @@ RUN /openhands/miniforge3/bin/mamba install conda-forge::poetry python=3.11 -y
...
     /openhands/miniforge3/bin/mamba run -n base poetry env use python3.11 && \
+    # Install project dependencies
     /openhands/miniforge3/bin/mamba run -n base poetry install --only main,runtime --no-interaction --no-root && \
+    # Update and install additional tools
     apt-get update && \
     /openhands/miniforge3/bin/mamba run -n base poetry run pip install playwright && \
     /openhands/miniforge3/bin/mamba run -n base poetry run playwright install --with-deps chromium && \
-    export OD_INTERPRETER_PATH=$(/openhands/miniforge3/bin/mamba run -n base poetry run python -c "import sys; print(sys.executable)") && \
+    # Set environment variables
+    export OH_INTERPRETER_PATH=$(/openhands/miniforge3/bin/mamba run -n base poetry run python -c "import sys; print(sys.executable)") && \
```
Comment (xingyaoww) [inline openhands/runtime/utils/runtime_templates/Dockerfile.j2:69]: Did we finish an global search/replace for `OD_INTERPRETER_PATH`?

**Review by xingyaoww — COMMENTED** [review]
LGTM! Once we are sure that the env var rename (od->oh) got replaced globally and all tests passed - i think this is good to go!

File: openhands/runtime/utils/runtime_templates/Dockerfile.j2 (hunk 1 above)
Comment (tobitege) [reply openhands/runtime/utils/runtime_templates/Dockerfile.j2:69]: No, I took this out from another PR test, I forgot to change it back to "OD_" for this PR.
I'll fix that with next commit, thanks!

Comment (tobitege) [reply openhands/runtime/utils/runtime_templates/Dockerfile.j2:69]: Replacements done now.

**Comment by tobitege** [comment]
Ohh, it seems this change is wrongly switching the Python path to the mamba folder and not the poetry folder.
Will investigate/fix this later today.

**Review by li-boxuan — APPROVED** [review]

**Pull request merged**